awx/ui-next/Makefile

```diff
@@ -0,0 +1,87 @@
+ui_next_mkfile_path := $(abspath $(lastword $(MAKEFILE_LIST)))
+ui_next_dir := $(dir $(ui_next_mkfile_path))
+
+ifeq ($(current_dir), undefined)
+	ui_next_dir := $(dir $(ui_next_mkfile_path))
+else
+	ui_next_dir := $(shell realpath --relative-to $(current_dir) $(ui_next_dir))/
+endif
+
+ANSIBLE_UI_GIT_REPO_HTTPS ?= https://github.com/ansible/ansible-ui.git
+ANSIBLE_UI_GIT_REPO_SSH ?= git@github.com:ansible/ansible-ui.git
+
+ANSIBLE_UI_GIT_BRANCH ?= main
+
+ANSIBLE_UI_LOCAL ?=
+
+ANSIBLE_UI_DIR ?= $(ui_next_dir)ansible-ui
```

Comment:
I don't love how this `ANSIBLE_UI_DIR` is handled. We can fix it up later, though. I would probably do something like this, though: `ANSIBLE_UI_URI` (a cloneable ui joint--e.g., a remote git repo, but could also be a local one!) and `ANSIBLE_UI_SRCDIR` could be something set irrespective of `ANSIBLE_UI_URI` (like if you already have an exploded copy, an sdist (without a `.git` dir! remember sdists don't have that!) or anything with that useable code in it). `ANSIBLE_UI_URI` could be populated after the clone when you need to clone or it could be set to the sdist dir or whatever.

Reply:
ur right!